crates/storage/provider/src/providers/snapshot/manager.rs

```diff
@@ -4,21 +4,34 @@ use dashmap::DashMap;
 use reth_interfaces::RethResult;
 use reth_nippy_jar::NippyJar;
 use reth_primitives::{
-    snapshot::BLOCKS_PER_SNAPSHOT, Address, BlockHash, BlockHashOrNumber, BlockNumber, ChainInfo,
-    Header, SealedHeader, SnapshotSegment, TransactionMeta, TransactionSigned,
-    TransactionSignedNoHash, TxHash, TxNumber, B256, U256,
+    snapshot::{HighestSnapshots, BLOCKS_PER_SNAPSHOT},
+    Address, BlockHash, BlockHashOrNumber, BlockNumber, ChainInfo, Header, SealedHeader,
+    SnapshotSegment, TransactionMeta, TransactionSigned, TransactionSignedNoHash, TxHash, TxNumber,
+    B256, U256,
 };
 use std::{ops::RangeBounds, path::PathBuf};
+use tokio::sync::watch;
 
 /// SnapshotProvider
 #[derive(Debug, Default)]
 pub struct SnapshotProvider {
     /// Maintains a map which allows for concurrent access to different `NippyJars`, over different
     /// segments and ranges.
     map: DashMap<(BlockNumber, SnapshotSegment), LoadedJar>,
+    /// Tracks the latest and highest snapshot of every segment.
+    highest_tracker: Option<watch::Receiver<Option<HighestSnapshots>>>,
```

Comment:
can we reuse `HighestSnapshotsTracker` here?

Reply:
that would mean adding tokio as a dependency to primitives, and I didn't want to add that tbh

Reply:
> that would mean adding tokio as a dependency to primitives

sorry I don't understand, `HighestSnapshotsTracker` is in `reth-snapshot` crate that we already use as a dependency in `reth-provider`, no?